pkg/platform/runtime/buildexpression/merge/merge.go

```diff
@@ -91,24 +97,45 @@ func getJsonMinusRequirements(expr *buildexpression.BuildExpression) (map[string
 	if !ok {
 		return nil, errs.New("'let' key is not a JSON object")
 	}
-	deleteRequirements(&letMap)
+	deleteKey(&letMap, "requirements")
+	// TODO: the following shouldn't be needed after DX-1939.
+	sortLists(&letMap)
+	deleteKey(&letMap, "at_time")
 
 	return m, nil
 }
 
-// deleteRequirements recursively iterates over the given JSON map until it finds a "requirements"
-// key and deletes it and its value.
-func deleteRequirements(m *map[string]interface{}) bool {
+// deleteKey recursively iterates over the given JSON map until it finds the given key and deletes
+// it and its value.
+func deleteKey(m *map[string]interface{}, key string) bool {
```

Comment:
This should have some unit tests I think, especially for covering the recursive case.

Reply:
It's not tested directly, but it is tested indirectly by the existing unit tests. I suppose I can add a direct, explicit test.

Reply:
Yeah I'd prefer direct tests over indirect ones. Indirect testing is what integration tests are meant to cover.